### ISSUE TITLE
CrashLoopBackOff seen after deploy via helm

### ISSUE THREAD

**Issue by krancour** [deis/etcd#15]
I don't see anything particularly useful when I `k describe pod <pod>`:

```
Events:
  FirstSeen LastSeen    Count   From            SubobjectPath               Reason      Message
  ─────────   ────────    ───── ────            ─────────────             ──────      ───────
  29s       29s     1   {kubelet 172.17.8.100}  implicitly required container POD   Pulled      Container image "gcr.io/google_containers/pause:0.8.0" already present on machine
  29s       29s     1   {kubelet 172.17.8.100}  implicitly required container POD   Created     Created with docker id abc8d779455d
  29s       29s     1   {kubelet 172.17.8.100}  implicitly required container POD   Started     Started with docker id abc8d779455d
  29s       29s     1   {scheduler }                            Scheduled   Successfully assigned deis-etcd-1-d8wj6 to 172.17.8.100
  27s       27s     1   {kubelet 172.17.8.100}  spec.containers{deis-etcd-1}        Created     Created with docker id e6a70c4341c1
  27s       27s     1   {kubelet 172.17.8.100}  spec.containers{deis-etcd-1}        Started     Started with docker id e6a70c4341c1
  27s       18s     2   {kubelet 172.17.8.100}  spec.containers{deis-etcd-1}        Pulled      Successfully pulled image "quay.io/deisci/etcd:v2-alpha"
  18s       18s     1   {kubelet 172.17.8.100}  spec.containers{deis-etcd-1}        Created     Created with docker id 09911910ffc9
  18s       18s     1   {kubelet 172.17.8.100}  spec.containers{deis-etcd-1}        Started     Started with docker id 09911910ffc9
  10s       10s     1   {kubelet 172.17.8.100}  spec.containers{deis-etcd-1}        Backoff     Back-off restarting failed docker container
  29s       0s      3   {kubelet 172.17.8.100}  spec.containers{deis-etcd-1}        Pulling     Pulling image "quay.io/deisci/etcd:v2-alpha"
```

**Comment by rimusz** [comment]
me too see the same with etcd deployed via deis charts

**Comment by mboersma** [comment]
I think that's expected actually--@technosophos can confirm this behavior. There isn't an explicit way in Helm to ensure that the discovery service is up first, so everything starts in parallel. The actual etcd containers will restart once or twice until they succeed in contacting the discovery service and establishing a quorum. The "normal" startup I see when using the deis chart is that the etcd containers have restarted twice after `CrashLoopBackoff`s.

**Comment by krancour** [comment]
@mboersma that makes sense, except I'm not seeing it eventually recover.  I've even deleted the etcd pods numerous times and allowed the RC to re-create them.  Also, after stepping away for 13 hours, I think I saw a retry count in the high 20s (I assume backoff puts increasing amounts of time between retries) with no success.

**Comment by krancour** [comment]
I also considered the possibility that this was somehow related to https://github.com/deis/charts/pull/12, so I undid that to be sure and it made no difference.

The last thing I tried... I wondered if perhaps this was dependent on the DNS add on-- which I was not running at the time.  I added that add on back into the mix, also with no difference.

**Comment by arschles** [comment]
hmm, I saw this when I `helm install`ed this chart the first time, and it never recovered. after a `helm uninstall`, all subsequent `helm install`s worked fine and I can't repro this anymore. I am deploying onto a [micro-kube](https://github.com/krancour/micro-kube) cluster

**Comment by mboersma** [comment]
> except I'm not seeing it eventually recover

Well that's different then and sounds like something needs to be fixed. :-/

**Comment by krancour** [comment]
> after a helm uninstall, all subsequent helm installs worked fine

Tried this.  Didn't help.  :(

I wish I had more insightful logs to go on here.

**Comment by mboersma** [comment]
I'm running into this as well now, and it's difficult to debug since there are basically no relevant logs. @technosophos and I took a look at it but we need to add some debug output. It's worth noting that you can easily connect to the discovery endpoint from another container in the namespace using the `DEIS_ETCD_*` environment variables and `curl`.

**Comment by bacongobbler** [comment]
I've found a bunch of logs available inside my vagrant machine. I just had to log into the minion to see them: https://gist.github.com/bacongobbler/db6d23ffdef2350ad932

Some sample logs available: https://gist.github.com/bacongobbler/41a659d18795684b6b70

looks like port 4100 is already in use?

**Comment by technosophos** [comment]
Thanks to @helgi I think I know what the issue is, and am fixing it now.

**Comment by mboersma** [comment]
Another log: https://gist.github.com/mboersma/690c1facc0d2d18374c2

`etcdmain: listen tcp :4100: bind: address already in use`

**Comment by rimusz** [comment]
@technosophos have all fixes been merged? I use the latest `quay.io/deisci/etcd:v2-alpha` created recently, but still see the same errors there as before:

```
$ kubectl --namespace=deis logs deis-etcd-1-kaivl
[error] 2015/12/18 16:33:58 Failed aboutme.FromEnv: Get https://10.100.0.1:443/api/v1/namespaces/deis/pods/deis-etcd-1-kaivl: EOF
[debug] 2015/12/18 16:33:58 Name: DEIS_ETCD_1_SERVICE_HOST, Val: 10.100.217.55
[debug] 2015/12/18 16:33:58 Name: DEIS_ETCD_CLUSTER_SIZE, Val: 3
[debug] 2015/12/18 16:33:58 Name: DEIS_ETCD_DISCOVERY_TOKEN, Val: E2D64576-D547-498C-AF71-3CF6EF319A8B
[debug] 2015/12/18 16:33:58 Name: ETCD_LISTEN_PEER_URLS, Val: http://:2380
[debug] 2015/12/18 16:33:58 Name: ETCD_ADVERTISE_CLIENT_URLS, Val: http://:4100
[debug] 2015/12/18 16:33:58 Name: DEIS_ETCD_DISCOVERY_SERVICE_HOST, Val: 10.100.53.99
[debug] 2015/12/18 16:33:58 Name: DEIS_ETCD_DISCOVERY_SERVICE_PORT, Val: 2381
[debug] 2015/12/18 16:33:58 Name: ETCD_INITIAL_ADVERTISE_PEER_URLS, Val: http://:2380
[debug] 2015/12/18 16:33:58 Name: ETCD_LISTEN_CLIENT_URLS, Val: http://:4100,http://127.0.0.1:4100
[debug] 2015/12/18 16:33:58 Name: DEIS_ETCD_1_SERVICE_PORT_CLIENT, Val: 4100
[debug] 2015/12/18 16:33:58 Name: HOSTNAME, Val: deis-etcd-1-kaivl
[info] 2015/12/18 16:33:58 Client configured for Etcd servers 'http://10.100.53.99:2381'
[info] 2015/12/18 16:33:58 Client configured for Etcd servers 'http://10.100.217.55:4100'
[debug] 2015/12/18 16:33:58 Name: ETCD_DISCOVERY, Val: http://10.100.53.99:2381/v2/keys/deis/discovery/E2D64576-D547-498C-AF71-3CF6EF319A8B
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=deis-etcd-1-kaivl
DEIS_ETCD_CLUSTER_SIZE=3
ETCD_NAME=deis1
POD_NAMESPACE=deis
POD_NAME=deis-etcd-1-kaivl
DEIS_WORKFLOW_SERVICE_HOST=10.100.254.237
KUBERNETES_SERVICE_PORT=443
KUBERNETES_PORT=tcp://10.100.0.1:443
DEIS_ROUTER_PORT_80_TCP_PORT=80
DEIS_ROUTER_PORT_80_TCP_ADDR=10.100.115.42
DEIS_ROUTER_PORT_443_TCP=tcp://10.100.115.42:443
DEIS_ROUTER_PORT_9090_TCP=tcp://10.100.115.42:9090
DEIS_MINIO_PORT_9000_TCP_PORT=9000
DEIS_WORKFLOW_SERVICE_PORT=80
DEIS_REGISTRY_PORT_5000_TCP=tcp://10.100.208.222:5000
DEIS_ETCD_1_PORT_2380_TCP_PORT=2380
DEIS_ETCD_1_PORT_4100_TCP_ADDR=10.100.217.55
KUBERNETES_PORT_443_TCP=tcp://10.100.0.1:443
KUBERNETES_PORT_443_TCP_ADDR=10.100.0.1
DEIS_ETCD_DISCOVERY_SERVICE_PORT_CLIENT=2381
DEIS_DATABASE_PORT_5432_TCP_PROTO=tcp
DEIS_ROUTER_PORT_443_TCP_ADDR=10.100.115.42
DEIS_MINIO_SERVICE_HOST=10.100.204.205
DEIS_ETCD_1_PORT_2380_TCP_PROTO=tcp
DEIS_ETCD_DISCOVERY_PORT_2381_TCP_PROTO=tcp
DEIS_ROUTER_SERVICE_PORT_HTTP=80
DEIS_ROUTER_SERVICE_PORT_BUILDER=2222
DEIS_ROUTER_PORT_9090_TCP_ADDR=10.100.115.42
DEIS_MINIO_PORT=tcp://10.100.204.205:9000
DEIS_MINIO_PORT_9000_TCP=tcp://10.100.204.205:9000
DEIS_DATABASE_PORT_5432_TCP_ADDR=10.100.0.144
DEIS_ETCD_1_PORT_2380_TCP_ADDR=10.100.217.55
KUBERNETES_SERVICE_HOST=10.100.0.1
DEIS_ROUTER_PORT_443_TCP_PORT=443
DEIS_ROUTER_PORT_9090_TCP_PROTO=tcp
DEIS_MINIO_SERVICE_PORT_S3=9000
DEIS_REGISTRY_SERVICE_PORT_HTTP=5000
DEIS_REGISTRY_PORT_5000_TCP_PROTO=tcp
DEIS_BUILDER_SERVICE_HOST=10.100.80.31
DEIS_BUILDER_SERVICE_PORT=2222
DEIS_WORKFLOW_PORT_80_TCP_ADDR=10.100.254.237
DEIS_REGISTRY_SERVICE_HOST=10.100.208.222
DEIS_REGISTRY_SERVICE_PORT=5000
DEIS_REGISTRY_PORT_5000_TCP_PORT=5000
DEIS_REGISTRY_PORT_5000_TCP_ADDR=10.100.208.222
DEIS_ETCD_1_SERVICE_PORT_CLIENT=4100
DEIS_ETCD_DISCOVERY_SERVICE_HOST=10.100.53.99
DEIS_DATABASE_PORT=tcp://10.100.0.144:5432
DEIS_ETCD_1_SERVICE_PORT_PEER=2380
KUBERNETES_PORT_443_TCP_PORT=443
DEIS_ROUTER_PORT_2222_TCP_ADDR=10.100.115.42
DEIS_ETCD_1_PORT_4100_TCP_PROTO=tcp
KUBERNETES_PORT_443_TCP_PROTO=tcp
DEIS_BUILDER_SERVICE_PORT_SSH=2222
DEIS_DATABASE_SERVICE_HOST=10.100.0.144
DEIS_BUILDER_PORT_2222_TCP=tcp://10.100.80.31:2222
DEIS_ETCD_DISCOVERY_PORT_2381_TCP_PORT=2381
DEIS_ROUTER_PORT_2222_TCP=tcp://10.100.115.42:2222
DEIS_ROUTER_PORT_9090_TCP_PORT=9090
DEIS_MINIO_SERVICE_PORT=9000
DEIS_DATABASE_SERVICE_PORT=5432
DEIS_ETCD_1_PORT=tcp://10.100.217.55:2380
DEIS_ROUTER_SERVICE_HOST=10.100.115.42
DEIS_ROUTER_SERVICE_PORT=80
DEIS_ROUTER_PORT=tcp://10.100.115.42:80
DEIS_DATABASE_PORT_5432_TCP=tcp://10.100.0.144:5432
DEIS_ETCD_DISCOVERY_SERVICE_PORT=2381
DEIS_ETCD_DISCOVERY_PORT_2381_TCP_ADDR=10.100.53.99
DEIS_ROUTER_PORT_80_TCP_PROTO=tcp
DEIS_DATABASE_SERVICE_PORT_HTTP=5432
DEIS_ETCD_1_PORT_2380_TCP=tcp://10.100.217.55:2380
DEIS_ETCD_1_PORT_4100_TCP=tcp://10.100.217.55:4100
DEIS_BUILDER_PORT_2222_TCP_ADDR=10.100.80.31
DEIS_MINIO_PORT_9000_TCP_ADDR=10.100.204.205
DEIS_WORKFLOW_PORT=tcp://10.100.254.237:80
DEIS_DATABASE_PORT_5432_TCP_PORT=5432
DEIS_BUILDER_PORT=tcp://10.100.80.31:2222
DEIS_ROUTER_SERVICE_PORT_HTTPS=443
DEIS_ROUTER_SERVICE_PORT_HEALTHZ=9090
DEIS_ROUTER_PORT_2222_TCP_PORT=2222
DEIS_MINIO_PORT_9000_TCP_PROTO=tcp
DEIS_ETCD_1_SERVICE_PORT=2380
DEIS_ETCD_1_PORT_4100_TCP_PORT=4100
DEIS_BUILDER_PORT_2222_TCP_PORT=2222
DEIS_ETCD_DISCOVERY_PORT=tcp://10.100.53.99:2381
DEIS_ROUTER_PORT_80_TCP=tcp://10.100.115.42:80
DEIS_ROUTER_PORT_2222_TCP_PROTO=tcp
DEIS_WORKFLOW_PORT_80_TCP=tcp://10.100.254.237:80
DEIS_WORKFLOW_PORT_80_TCP_PROTO=tcp
DEIS_REGISTRY_PORT=tcp://10.100.208.222:5000
DEIS_ETCD_1_SERVICE_HOST=10.100.217.55
DEIS_BUILDER_PORT_2222_TCP_PROTO=tcp
DEIS_ETCD_DISCOVERY_PORT_2381_TCP=tcp://10.100.53.99:2381
DEIS_ROUTER_PORT_443_TCP_PROTO=tcp
DEIS_WORKFLOW_SERVICE_PORT_HTTP=80
DEIS_WORKFLOW_PORT_80_TCP_PORT=80
HOME=/root
MY_PORT_CLIENT=4100
MY_PORT_PEER=2380
DEIS_ETCD_DISCOVERY_TOKEN=E2D64576-D547-498C-AF71-3CF6EF319A8B
ETCD_LISTEN_PEER_URLS=http://:2380
ETCD_ADVERTISE_CLIENT_URLS=http://:4100
ETCD_INITIAL_ADVERTISE_PEER_URLS=http://:2380
ETCD_LISTEN_CLIENT_URLS=http://:4100,http://127.0.0.1:4100
ETCD_DISCOVERY=http://10.100.53.99:2381/v2/keys/deis/discovery/E2D64576-D547-498C-AF71-3CF6EF319A8B
2015-12-18 16:33:58.893004 I | flags: recognized and used environment variable ETCD_ADVERTISE_CLIENT_URLS=http://:4100
2015-12-18 16:33:58.893146 I | flags: recognized and used environment variable ETCD_DISCOVERY=http://10.100.53.99:2381/v2/keys/deis/discovery/E2D64576-D547-498C-AF71-3CF6EF319A8B
2015-12-18 16:33:58.893170 I | flags: recognized and used environment variable ETCD_INITIAL_ADVERTISE_PEER_URLS=http://:2380
2015-12-18 16:33:58.893194 I | flags: recognized and used environment variable ETCD_LISTEN_CLIENT_URLS=http://:4100,http://127.0.0.1:4100
2015-12-18 16:33:58.893204 I | flags: recognized and used environment variable ETCD_LISTEN_PEER_URLS=http://:2380
2015-12-18 16:33:58.893217 I | flags: recognized and used environment variable ETCD_NAME=deis1
2015-12-18 16:33:58.893335 I | etcdmain: etcd Version: 2.2.1
2015-12-18 16:33:58.893362 I | etcdmain: Git SHA: 75f8282
2015-12-18 16:33:58.893368 I | etcdmain: Go Version: go1.5.1
2015-12-18 16:33:58.893373 I | etcdmain: Go OS/Arch: linux/amd64
2015-12-18 16:33:58.893388 I | etcdmain: setting maximum number of CPUs to 2, total number of available CPUs is 2
2015-12-18 16:33:58.893397 W | etcdmain: no data-dir provided, using default data-dir ./deis1.etcd
2015-12-18 16:33:58.893503 I | etcdmain: listening for peers on http://:2380
2015-12-18 16:33:58.893560 I | etcdmain: listening for client requests on http://127.0.0.1:4100
2015-12-18 16:33:58.893603 I | etcdmain: stopping listening for client requests on http://127.0.0.1:4100
2015-12-18 16:33:58.893615 I | etcdmain: stopping listening for peers on http://:2380
2015-12-18 16:33:58.893629 C | etcdmain: listen tcp :4100: bind: address already in use
[error] 2015/12/18 16:33:58 Etcd quit unexpectedly: exit status 1
```

**Comment by technosophos** [comment]
There was an issue with the Quay image (it didn't get updated). We're re-pushing the image right now. If you re-test in about 15 minutes, it should be working as explained.

**Comment by rimusz** [comment]
cool I will give it go a bit later and report back :)

**Comment by technosophos** [comment]
Looking at the output above, I see an error I've not seen before:

`[error] 2015/12/18 16:33:58 Failed aboutme.FromEnv: Get https://10.100.0.1:443/api/v1/namespaces/deis/pods/deis-etcd-1-kaivl: EOF
`

I wonder what's causing that. It's a failure to connect to the kube API server.

**Comment by rimusz** [comment]
let's see what happens with the latest release

**Comment by jchauncey** [comment]
ok so new etcd image does not seem to include the fix we need

**Comment by jchauncey** [comment]
``` console
[error] 2015/12/18 17:35:59 Failed aboutme.FromEnv: Get https://10.100.0.1:443/api/v1/namespaces/deis/pods/deis-etcd-1-2ykqc: EOF
[debug] 2015/12/18 17:35:59 Name: DEIS_ETCD_CLUSTER_SIZE, Val: 3
[debug] 2015/12/18 17:35:59 Name: DEIS_ETCD_DISCOVERY_TOKEN, Val: E2D64576-D547-498C-AF71-3CF6EF319A8B
[debug] 2015/12/18 17:35:59 Name: HOSTNAME, Val: deis-etcd-1-2ykqc
[debug] 2015/12/18 17:35:59 Name: ETCD_LISTEN_PEER_URLS, Val: http://:2380
[debug] 2015/12/18 17:35:59 Name: DEIS_ETCD_1_SERVICE_HOST, Val: 10.100.93.125
[debug] 2015/12/18 17:35:59 Name: DEIS_ETCD_1_SERVICE_PORT_CLIENT, Val: 4100
[debug] 2015/12/18 17:35:59 Name: ETCD_INITIAL_ADVERTISE_PEER_URLS, Val: http://:2380
[debug] 2015/12/18 17:35:59 Name: ETCD_LISTEN_CLIENT_URLS, Val: http://:4100,http://127.0.0.1:4100
[debug] 2015/12/18 17:35:59 Name: ETCD_ADVERTISE_CLIENT_URLS, Val: http://:4100
[debug] 2015/12/18 17:35:59 Name: DEIS_ETCD_DISCOVERY_SERVICE_HOST, Val: 10.100.125.212
[debug] 2015/12/18 17:35:59 Name: DEIS_ETCD_DISCOVERY_SERVICE_PORT, Val: 2381
[info] 2015/12/18 17:35:59 Client configured for Etcd servers 'http://10.100.125.212:2381'
[info] 2015/12/18 17:35:59 Client configured for Etcd servers 'http://10.100.93.125:4100'
[debug] 2015/12/18 17:35:59 Name: ETCD_DISCOVERY, Val: http://10.100.125.212:2381/v2/keys/deis/discovery/E2D64576-D547-498C-AF71-3CF6EF319A8B
```

**Comment by technosophos** [comment]
@rimusz Sorry. There was a little comedy of errors here where I broke the travis build _and_ somehow rebased away a line of the patch. But the version that's currently getting built by Travis should solve the problem.

**Comment by technosophos** [comment]
@rimusz I'll wait to close until you verify that it works for you.

**Comment by jchauncey** [comment]
im waiting for master to build and then ill do a full deploy and make sure it all works

**Comment by jchauncey** [comment]
``` console
╰─± k get --namespace=deis po
NAME                        READY     STATUS    RESTARTS   AGE
deis-builder-6wewq          1/1       Running   0          53s
deis-database-njgee         1/1       Running   0          53s
deis-etcd-1-6cvs0           1/1       Running   0          53s
deis-etcd-1-leo8d           1/1       Running   0          53s
deis-etcd-1-vk8ps           1/1       Running   0          53s
deis-etcd-discovery-2dx6x   1/1       Running   0          53s
deis-minio-nc4um            0/1       Pending   0          52s
deis-registry-ddw9t         1/1       Running   0          52s
deis-router-h585i           1/1       Running   0          52s
deis-workflow-aypva         0/1       Pending   0          52s
```

**Comment by jchauncey** [comment]
looks good

**Comment by rimusz** [comment]
etcd x 3 looks good:

```
$ kubectl --namespace=deis get po
NAME                        READY     STATUS    RESTARTS   AGE
deis-builder-4a8zw          1/1       Running   1          5m
deis-database-kxkyz         1/1       Running   0          5m
deis-etcd-1-t5ho8           1/1       Running   0          5m
deis-etcd-1-u1zzi           1/1       Running   0          5m
deis-etcd-1-ysxmf           1/1       Running   0          5m
deis-etcd-discovery-fptle   1/1       Running   0          5m
deis-minio-a5cmb            1/1       Running   0          5m
deis-registry-iai4e         1/1       Running   0          5m
deis-router-9spey           1/1       Running   0          5m
deis-workflow-tt1oe         1/1       Running   0          5m
```

**Comment by rimusz** [comment]
changed cluster size and replicas to 1, not so good (found 1 peer(s), waiting for 2 more):

```
$ kubectl --namespace=deis logs deis-etcd-1-i3ywx
[error] 2015/12/18 20:44:34 Failed aboutme.FromEnv: Get https://10.100.0.1:443/api/v1/namespaces/deis/pods/deis-etcd-1-i3ywx: EOF
[warning] 2015/12/18 20:44:34 Attempting to recover.
[warning] 2015/12/18 20:44:34 No IP found by API query.
[debug] 2015/12/18 20:44:35 Name: DEIS_ETCD_DISCOVERY_SERVICE_HOST, Val: 10.100.182.228
[debug] 2015/12/18 20:44:35 Name: DEIS_ETCD_DISCOVERY_SERVICE_PORT, Val: 2381
[debug] 2015/12/18 20:44:35 Name: DEIS_ETCD_CLUSTER_SIZE, Val: 1
[debug] 2015/12/18 20:44:35 Name: HOSTNAME, Val: deis-etcd-1-i3ywx
[debug] 2015/12/18 20:44:35 Name: DEIS_ETCD_1_SERVICE_HOST, Val: 10.100.104.240
[debug] 2015/12/18 20:44:35 Name: DEIS_ETCD_1_SERVICE_PORT_CLIENT, Val: 4100
[debug] 2015/12/18 20:44:35 Name: DEIS_ETCD_DISCOVERY_TOKEN, Val: E2D64576-D547-498C-AF71-3CF6EF319A8B
[debug] 2015/12/18 20:44:35 Name: ETCD_LISTEN_PEER_URLS, Val: http://10.244.7.8:2380
[debug] 2015/12/18 20:44:35 Name: ETCD_INITIAL_ADVERTISE_PEER_URLS, Val: http://10.244.7.8:2380
[debug] 2015/12/18 20:44:35 Name: ETCD_LISTEN_CLIENT_URLS, Val: http://10.244.7.8:4100,http://127.0.0.1:4100
[debug] 2015/12/18 20:44:35 Name: ETCD_ADVERTISE_CLIENT_URLS, Val: http://10.244.7.8:4100
[info] 2015/12/18 20:44:35 Client configured for Etcd servers 'http://10.100.182.228:2381'
[info] 2015/12/18 20:44:35 Client configured for Etcd servers 'http://10.100.104.240:4100'
[debug] 2015/12/18 20:44:35 Name: ETCD_DISCOVERY, Val: http://10.100.182.228:2381/v2/keys/deis/discovery/E2D64576-D547-498C-AF71-3CF6EF319A8B
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=deis-etcd-1-i3ywx
DEIS_ETCD_CLUSTER_SIZE=1
ETCD_NAME=deis-etcd-1-i3ywx
POD_NAMESPACE=deis
POD_NAME=deis-etcd-1-i3ywx
DEIS_WORKFLOW_SERVICE_PORT_HTTP=80
DEIS_DATABASE_PORT_5432_TCP=tcp://10.100.171.70:5432
DEIS_BUILDER_SERVICE_PORT=2222
DEIS_MINIO_SERVICE_PORT=9000
DEIS_ROUTER_SERVICE_PORT_HTTP=80
DEIS_ROUTER_PORT_2222_TCP_ADDR=10.100.108.111
DEIS_ETCD_DISCOVERY_SERVICE_PORT_CLIENT=2381
DEIS_BUILDER_SERVICE_HOST=10.100.43.164
DEIS_BUILDER_PORT=tcp://10.100.43.164:2222
DEIS_ROUTER_PORT_80_TCP_ADDR=10.100.108.111
KUBERNETES_PORT_443_TCP=tcp://10.100.0.1:443
DEIS_BUILDER_PORT_2222_TCP_PROTO=tcp
DEIS_MINIO_PORT_9000_TCP_ADDR=10.100.152.187
DEIS_ETCD_1_PORT_4100_TCP_PROTO=tcp
DEIS_ROUTER_SERVICE_PORT=80
DEIS_WORKFLOW_SERVICE_PORT=80
DEIS_WORKFLOW_PORT_80_TCP=tcp://10.100.240.27:80
DEIS_REGISTRY_PORT=tcp://10.100.83.52:5000
DEIS_ROUTER_PORT_80_TCP=tcp://10.100.108.111:80
DEIS_ROUTER_PORT_80_TCP_PROTO=tcp
DEIS_ROUTER_PORT_443_TCP_ADDR=10.100.108.111
DEIS_ROUTER_PORT_9090_TCP_ADDR=10.100.108.111
DEIS_ETCD_1_PORT_2380_TCP_ADDR=10.100.104.240
DEIS_ETCD_DISCOVERY_PORT_2381_TCP_PORT=2381
KUBERNETES_SERVICE_HOST=10.100.0.1
DEIS_WORKFLOW_SERVICE_HOST=10.100.240.27
DEIS_DATABASE_SERVICE_HOST=10.100.171.70
DEIS_DATABASE_PORT_5432_TCP_ADDR=10.100.171.70
DEIS_ETCD_1_SERVICE_PORT_PEER=2380
DEIS_ETCD_1_PORT_2380_TCP_PROTO=tcp
DEIS_REGISTRY_PORT_5000_TCP=tcp://10.100.83.52:5000
DEIS_REGISTRY_PORT_5000_TCP_ADDR=10.100.83.52
KUBERNETES_PORT_443_TCP_PROTO=tcp
DEIS_DATABASE_SERVICE_PORT_HTTP=5432
DEIS_ROUTER_PORT_9090_TCP=tcp://10.100.108.111:9090
DEIS_ETCD_DISCOVERY_PORT=tcp://10.100.182.228:2381
KUBERNETES_PORT_443_TCP_PORT=443
DEIS_BUILDER_PORT_2222_TCP=tcp://10.100.43.164:2222
DEIS_MINIO_PORT_9000_TCP=tcp://10.100.152.187:9000
DEIS_ETCD_1_SERVICE_PORT=2380
DEIS_ROUTER_PORT_80_TCP_PORT=80
DEIS_DATABASE_PORT=tcp://10.100.171.70:5432
DEIS_ETCD_1_PORT_2380_TCP_PORT=2380
DEIS_ROUTER_PORT_443_TCP_PORT=443
DEIS_ETCD_DISCOVERY_SERVICE_PORT=2381
DEIS_BUILDER_SERVICE_PORT_SSH=2222
DEIS_ETCD_1_SERVICE_HOST=10.100.104.240
DEIS_ROUTER_SERVICE_PORT_HTTPS=443
DEIS_ROUTER_PORT_2222_TCP=tcp://10.100.108.111:2222
DEIS_MINIO_SERVICE_HOST=10.100.152.187
DEIS_MINIO_SERVICE_PORT_S3=9000
DEIS_ETCD_1_PORT=tcp://10.100.104.240:2380
DEIS_ROUTER_PORT_9090_TCP_PROTO=tcp
DEIS_ETCD_DISCOVERY_PORT_2381_TCP_ADDR=10.100.182.228
DEIS_DATABASE_PORT_5432_TCP_PORT=5432
DEIS_ETCD_1_PORT_4100_TCP_PORT=4100
DEIS_ROUTER_SERVICE_PORT_BUILDER=2222
DEIS_ETCD_DISCOVERY_PORT_2381_TCP=tcp://10.100.182.228:2381
DEIS_WORKFLOW_PORT=tcp://10.100.240.27:80
DEIS_WORKFLOW_PORT_80_TCP_PORT=80
DEIS_ETCD_1_PORT_4100_TCP=tcp://10.100.104.240:4100
DEIS_ETCD_1_PORT_4100_TCP_ADDR=10.100.104.240
DEIS_ROUTER_SERVICE_HOST=10.100.108.111
DEIS_ROUTER_PORT_9090_TCP_PORT=9090
KUBERNETES_SERVICE_PORT=443
DEIS_BUILDER_PORT_2222_TCP_ADDR=10.100.43.164
DEIS_REGISTRY_SERVICE_PORT=5000
DEIS_REGISTRY_SERVICE_PORT_HTTP=5000
DEIS_ROUTER_SERVICE_PORT_HEALTHZ=9090
DEIS_ROUTER_PORT_443_TCP=tcp://10.100.108.111:443
DEIS_ETCD_DISCOVERY_PORT_2381_TCP_PROTO=tcp
KUBERNETES_PORT_443_TCP_ADDR=10.100.0.1
DEIS_DATABASE_PORT_5432_TCP_PROTO=tcp
DEIS_MINIO_PORT=tcp://10.100.152.187:9000
DEIS_ETCD_1_SERVICE_PORT_CLIENT=4100
DEIS_REGISTRY_PORT_5000_TCP_PORT=5000
DEIS_ROUTER_PORT_2222_TCP_PROTO=tcp
DEIS_ETCD_DISCOVERY_SERVICE_HOST=10.100.182.228
DEIS_DATABASE_SERVICE_PORT=5432
DEIS_MINIO_PORT_9000_TCP_PROTO=tcp
DEIS_MINIO_PORT_9000_TCP_PORT=9000
DEIS_ETCD_1_PORT_2380_TCP=tcp://10.100.104.240:2380
DEIS_REGISTRY_PORT_5000_TCP_PROTO=tcp
DEIS_ROUTER_PORT=tcp://10.100.108.111:80
KUBERNETES_PORT=tcp://10.100.0.1:443
DEIS_WORKFLOW_PORT_80_TCP_PROTO=tcp
DEIS_WORKFLOW_PORT_80_TCP_ADDR=10.100.240.27
DEIS_BUILDER_PORT_2222_TCP_PORT=2222
DEIS_REGISTRY_SERVICE_HOST=10.100.83.52
DEIS_ROUTER_PORT_443_TCP_PROTO=tcp
DEIS_ROUTER_PORT_2222_TCP_PORT=2222
HOME=/root
MY_IP=10.244.7.8
MY_NODEIP=
MY_NAMESPACE=deis
MY_SELFLINK=
MY_UID=
MY_APISERVER=https://10.100.0.1:443
MY_NAME=deis-etcd-1-i3ywx
MY_PORT_CLIENT=4100
MY_PORT_PEER=2380
DEIS_ETCD_DISCOVERY_TOKEN=E2D64576-D547-498C-AF71-3CF6EF319A8B
ETCD_LISTEN_PEER_URLS=http://10.244.7.8:2380
ETCD_INITIAL_ADVERTISE_PEER_URLS=http://10.244.7.8:2380
ETCD_LISTEN_CLIENT_URLS=http://10.244.7.8:4100,http://127.0.0.1:4100
ETCD_ADVERTISE_CLIENT_URLS=http://10.244.7.8:4100
ETCD_DISCOVERY=http://10.100.182.228:2381/v2/keys/deis/discovery/E2D64576-D547-498C-AF71-3CF6EF319A8B
2015-12-18 20:44:35.026016 I | flags: recognized and used environment variable ETCD_ADVERTISE_CLIENT_URLS=http://10.244.7.8:4100
2015-12-18 20:44:35.026107 I | flags: recognized and used environment variable ETCD_DISCOVERY=http://10.100.182.228:2381/v2/keys/deis/discovery/E2D64576-D547-498C-AF71-3CF6EF319A8B
2015-12-18 20:44:35.026122 I | flags: recognized and used environment variable ETCD_INITIAL_ADVERTISE_PEER_URLS=http://10.244.7.8:2380
2015-12-18 20:44:35.026132 I | flags: recognized and used environment variable ETCD_LISTEN_CLIENT_URLS=http://10.244.7.8:4100,http://127.0.0.1:4100
2015-12-18 20:44:35.026137 I | flags: recognized and used environment variable ETCD_LISTEN_PEER_URLS=http://10.244.7.8:2380
2015-12-18 20:44:35.026145 I | flags: recognized and used environment variable ETCD_NAME=deis-etcd-1-i3ywx
2015-12-18 20:44:35.026210 I | etcdmain: etcd Version: 2.2.1
2015-12-18 20:44:35.026223 I | etcdmain: Git SHA: 75f8282
2015-12-18 20:44:35.026228 I | etcdmain: Go Version: go1.5.1
2015-12-18 20:44:35.026256 I | etcdmain: Go OS/Arch: linux/amd64
2015-12-18 20:44:35.026263 I | etcdmain: setting maximum number of CPUs to 2, total number of available CPUs is 2
2015-12-18 20:44:35.026268 W | etcdmain: no data-dir provided, using default data-dir ./deis-etcd-1-i3ywx.etcd
2015-12-18 20:44:35.026347 I | etcdmain: listening for peers on http://10.244.7.8:2380
2015-12-18 20:44:35.026389 I | etcdmain: listening for client requests on http://10.244.7.8:4100
2015-12-18 20:44:35.026407 I | etcdmain: listening for client requests on http://127.0.0.1:4100
2015-12-18 20:44:35.035165 N | discovery: found self ed351d21eadeedb2 in the cluster
2015-12-18 20:44:35.035196 N | discovery: found 1 peer(s), waiting for 2 more
```

**Comment by rimusz** [comment]
I would love to see support for 1 size cluster, as that would make easier for demos using my kube-solo 
with stateful Deis-lite cluster :)

**Comment by rimusz** [comment]
stupid me, forgot to change discovery rc

**Comment by rimusz** [comment]
cool, cluster size 1 works too, phew :)

only this error still shows up at both 1 and 3 node clusters:

```
[error] 2015/12/18 20:44:34 Failed aboutme.FromEnv: Get https://10.100.0.1:443/api/v1/namespaces/deis/pods/deis-etcd-1-i3ywx: EOF
[warning] 2015/12/18 20:44:34 Attempting to recover.
[warning] 2015/12/18 20:59:57 No IP found by API query.
```

**Comment by rimusz** [comment]
holly shit even reloading kube-solo Deis cluster survived :)
AWESOME !!!

**Comment by technosophos** [comment]
So this error is something I'd like to understand:

`[error] 2015/12/18 20:44:34 Failed aboutme.FromEnv: Get https://10.100.0.1:443/api/v1/namespaces/deis/pods/deis-etcd-1-i3ywx: EOF
`

I'm going to open a separate issue for that, and mark this one as closed.
